### PR TITLE
Enhancements for VoiceChannel join methods

### PR DESCRIPTION
**Pull request by 5antos**
Added options to choose if the bot connects to a voice channel being muted or deafened instead of always joining without being muted and deafened.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -1756,7 +1756,7 @@ declare namespace Eris {
     getVoiceRegions(guildID?: string): Promise<VoiceRegion[]>;
     getWebhook(webhookID: string, token?: string): Promise<Webhook>;
     getWebhookMessage(webhookID: string, token: string, messageID: string): Promise<Message<GuildTextableChannel>>;
-    joinVoiceChannel(channelID: string, options?: { opusOnly?: boolean; shared?: boolean }): Promise<VoiceConnection>;
+    joinVoiceChannel(channelID: string, options?: { opusOnly?: boolean; shared?: boolean, selfMute?: boolean, selfDeaf?: boolean }): Promise<VoiceConnection>;
```
Comment (almeidx) [inline index.d.ts]:
```suggestion
    joinVoiceChannel(channelID: string, options?: { opusOnly?: boolean; shared?: boolean; selfMute?: boolean; selfDeaf?: boolean }): Promise<VoiceConnection>;
```

File: index.d.ts [hunk 2]
```diff
@@ -2593,7 +2593,7 @@ declare namespace Eris {
     voiceMembers: Collection<Member>;
     createInvite(options?: CreateInviteOptions, reason?: string): Promise<Invite<"withMetadata", VoiceChannel>>;
     getInvites(): Promise<(Invite<"withMetadata", VoiceChannel>)[]>;
-    join(options: { opusOnly?: boolean; shared?: boolean }): Promise<VoiceConnection>;
+    join(options: { opusOnly?: boolean; shared?: boolean, selfMute?: boolean, selfDeaf?: boolean }): Promise<VoiceConnection>;
```
Comment (almeidx) [inline index.d.ts]:
```suggestion
    join(options: { opusOnly?: boolean; shared?: boolean; selfMute?: boolean; selfDeaf?: boolean }): Promise<VoiceConnection>;
```

**Review by almeidx — CHANGES_REQUESTED** [review]

File: index.d.ts [hunk 3]
```diff
@@ -1756,7 +1756,7 @@ declare namespace Eris {
     getVoiceRegions(guildID?: string): Promise<VoiceRegion[]>;
     getWebhook(webhookID: string, token?: string): Promise<Webhook>;
     getWebhookMessage(webhookID: string, token: string, messageID: string): Promise<Message<GuildTextableChannel>>;
-    joinVoiceChannel(channelID: string, options?: { opusOnly?: boolean; shared?: boolean }): Promise<VoiceConnection>;
+    joinVoiceChannel(channelID: string, options?: { opusOnly?: boolean; shared?: boolean; selfMute?: boolean; selfDeaf?: boolean }): Promise<VoiceConnection>;
```
Comment (bsian03) [inline index.d.ts]: Can this and the same one at L2596 be in its own interface?

**Review by bsian03 — CHANGES_REQUESTED** [review]

**Comment by bsian03** [comment]
Any update on this?

**Comment by JustCat80** [comment]
~~If there are no replies within a day or so I'll assume this is abandoned and pick it up myself~~
Nvm bsian said he'll do it :)

**Review by bsian03 — DISMISSED** [review]

**Comment by bsian03** [comment]
Flipping conflict

**Review by bsian03 — APPROVED** [review]

**Pull request merged**